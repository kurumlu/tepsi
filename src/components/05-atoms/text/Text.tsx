import { FunctionComponent, createElement } from 'react';
import cn from 'classnames';

import { GLOSSARY_HIGHLIGHT_CLASS } from '../../00-helpers/constants/GlossaryConstants';
import { ItemProp } from '../../00-helpers/enums/ItemProp';

export enum TextColor {
  NEGATIVE = 'negative',
  POSITIVE = 'positive',
  SUBTLE = 'subtle',
  DISCOUNT = 'discount',
  INHERIT = 'inherit',
  INFO = 'info',
  NOTICE = 'notice',
}

export enum TextSize {
  XLARGE = 'xl',
  LARGE = 'l',
  MEDIUM = 'm',
  SMALL = 's',
  XSMALL = 'xs',
  INHERIT = 'none',
}

export enum TextType {
  EXTRA_LARGE_CONTENT = 'extra-large-content',
  EXTRA_CONTENT = 'extra-content',
  CONTENT = 'content',
  SUB_CONTENT = 'sub-content',
  TITLE = 'title',
}

export enum TextAlign {
  CENTER = 'center',
  RIGHT = 'right',
}

export enum TextSemanticTag {
  DEL = 'del',
  PARAGRAPH = 'p',
  STRONG = 'strong',
  SPAN = 'span',
  DIV = 'div',
}

export enum TextFeature {
  PRODUCT_TECHNICAL_FEATURES_ODD = 'product-technical-features-odd',
  PRODUCT_TECHNICAL_FEATURES_EVEN = 'product-technical-features-even',
  PRODUCT_DETAILS = 'product-details',
  PRODUCT_TECHNICAL_DESCRIPTION = 'product-technical-description',
}

export type TextProps = {
  size?: TextSize;
  type?: TextType;
  color?: TextColor;
  strike?: boolean;
  paragraph?: boolean;
  emphasized?: boolean;
  align?: TextAlign;
  className?: string;
  dataQA?: string;
  withGlossary?: boolean;
  feature?: TextFeature;
  isClipped?: boolean;
  isUppercase?: boolean;
  inline?: boolean;
  html?: string;
  lineClamp?: 1 | 2 | 3 | 4 | 5;
  itemProp?: ItemProp;
  itemContent?: string;
};

export const Text: FunctionComponent<TextProps> = ({
  size,
  type,
  color,
  strike,
  paragraph,
  emphasized,
  align,
  className,
  children,
  dataQA,
  withGlossary,
  feature,
  isClipped,
  isUppercase,
  inline,
  html,
  lineClamp,
  itemProp,
  itemContent,
}) => {
  const getType = () =>
    paragraph
      ? TextSemanticTag.PARAGRAPH
      : strike && emphasized
      ? TextSemanticTag.STRONG
      : inline
      ? TextSemanticTag.SPAN
      : TextSemanticTag.DIV;

  const getClassName = () =>
    cn(
      'as-a-text',
      {
        [`as-a-text--${color}`]: color,
        [`as-a-text--${size}`]: size,
        [`as-a-text--${type}`]: type,
        [`as-a-text--${align}`]: align,
        [`as-a-text--${feature}`]: feature,
        [`as-a-text--line-clamp-${lineClamp}`]: lineClamp,
        [GLOSSARY_HIGHLIGHT_CLASS]: withGlossary,
        'as-a-text--html': html,
        'as-a-text--clipped': isClipped,
        'as-a-text--uppercase': isUppercase,
      },
      className
    );

  const getProps = () => ({
    className: getClassName(),
    'data-qa': dataQA,
    itemProp,
    content: itemContent,
  });

  const getChildren = () =>
    strike || emphasized
      ? createElement(
          strike ? TextSemanticTag.DEL : TextSemanticTag.STRONG,
          null,
          children
        )
      : children;

  if (html) {
    return createElement(getType(), {
      ...getProps(),
      dangerouslySetInnerHTML: {
        __html: html,
      },
    });
  }

  return createElement(getType(), getProps(), getChildren());
};
