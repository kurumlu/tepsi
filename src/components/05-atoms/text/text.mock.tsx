import * as React from 'react';
import { FunctionComponent } from 'react';

export const mockText =
  'We believe there’s no substitute for experience. That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.';

export const MockChildren: FunctionComponent = () => (
  <ul>
    <li>Test text 1</li>
    <li>Test text 2</li>
    <li>Test text 3</li>
  </ul>
);

export const mockHtml =
  "<div class='sd-a-text--test-classname'>Some <strong>HTML</strong> text";
