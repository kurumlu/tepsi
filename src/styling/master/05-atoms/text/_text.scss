// Variables for Text ATOM
$a-text--color: $t2-color__foreground !default;
$a-text--font-size: $t2-font-size__body-m !default;
$a-text--line-height: $t2-line-height__m !default;
$a-text--font-map: $t2-font-map__body !default;

// --size
$a-text--xs--font-size: $t2-font-size__body-xs !default;
$a-text--xs--line-height: $t2-line-height__xxs !default;

$a-text--s--font-size: $t2-font-size__body-s !default;
$a-text--s--line-height: $t2-line-height__s !default;

$a-text--l--font-size: $t2-font-size__body-l !default;
$a-text--l--line-height: $t2-line-height__l !default;

$a-text--xl--font-size: $t2-font-size__body-xl !default;
$a-text--xl--line-height: $t2-line-height__xxs !default;

// --color
$a-text--subtle--color: $t2-color__subtle !default;
$a-text--positive--color: $t2-color__positive !default;
$a-text--negative--color: $t2-color__negative !default;
$a-text--discount--color: $t2-color__discount !default;
$a-text--info--color: $t2-color__data !default;
$a-text--notice--color: get-ui-color(o, light) !default;

// --product-technical-features section
$a-text--product-technical-features--padding: get-spacing(1.5) 0 get-spacing(1.5) get-spacing(3) !default;
$a-text--product-technical-features--mobile--padding: get-spacing(1.5) get-spacing(2) !default;
$a-text--product-technical-features__li--nth-child--background: $t2-color__gray-10 !default;

// --product-details
$a-text--product-details__li--margin-bottom: get-spacing(1) !default;

// --product-technical-description
$a-text--product-technical-description__ul--padding-left: get-spacing(2) !default;
$a-text--product-technical-description__li--margin-bottom: get-spacing(2) !default;

// Code for Text ATOM
.as-a-text {
  color: $a-text--color;
  font-size: $a-text--font-size;
  line-height: $a-text--line-height;

  @include font-map($a-text--font-map);

  text-decoration: none;
}

// --size
.as-a-text--xs {
  font-size: $a-text--xs--font-size;
  line-height: $a-text--xs--line-height;
}

.as-a-text--s {
  font-size: $a-text--s--font-size;
  line-height: $a-text--s--line-height;
}

.as-a-text--l {
  font-size: $a-text--l--font-size;
  line-height: $a-text--l--line-height;
}

.as-a-text--xl {
  font-size: $a-text--xl--font-size;
  line-height: $a-text--xl--line-height;
}

// --type
.as-a-text--extra-large-content {
  font-size: $a-text--s--font-size;
  line-height: $a-text--s--line-height;

  @include mq($from: tablet) {
    font-size: $a-text--font-size;
    line-height: $a-text--line-height;
  }

  @include mq($from: desktop) {
    font-size: $a-text--xl--font-size;
    line-height: $a-text--xl--line-height;
  }
}

.as-a-text--extra-content {
  font-size: $a-text--font-size;
  line-height: $a-text--line-height;

  @include mq($from: tablet) {
    font-size: $a-text--l--font-size;
    line-height: $a-text--l--line-height;
  }
}

.as-a-text--content {
  font-size: $a-text--s--font-size;
  line-height: $a-text--s--line-height;

  @include mq($from: tablet) {
    font-size: $a-text--font-size;
    line-height: $a-text--line-height;
  }
}

.as-a-text--sub-content {
  font-size: $a-text--xs--font-size;
  line-height: $a-text--xs--line-height;

  @include mq($from: tablet) {
    font-size: $a-text--s--font-size;
    line-height: $a-text--s--line-height;
  }
}

.as-a-text--title {
  font-size: $a-text--l--font-size;
  line-height: $a-text--l--line-height;

  @include mq($from: tablet) {
    font-size: $a-text--xl--font-size;
    line-height: $a-text--xl--line-height;
  }
}

.as-a-text--none {
  font-size: inherit;
  line-height: inherit;
}

// --color
.as-a-text--subtle {
  color: $a-text--subtle--color;
}

.as-a-text--positive {
  color: $a-text--positive--color;
}

.as-a-text--negative {
  color: $a-text--negative--color;
}

.as-a-text--discount {
  color: $a-text--discount--color;
}

.as-a-text--inherit {
  color: inherit;
}

.as-a-text--info {
  color: $a-text--info--color;
}

.as-a-text--notice {
  color: $a-text--notice--color;
}

// --align
.as-a-text--center {
  text-align: center;
}

.as-a-text--right {
  text-align: right;
}

// --uppercase
.as-a-text--uppercase {
  text-transform: uppercase;
}

// emphasized
strong.as-a-text,
.as-a-text strong {
  @include font-map(
    $map: $t2-font-map__body-emp,
    $style: inherit,
    $letter-spacing: inherit,
    $text-transform: inherit
  );
}

// clipped
.as-a-text--clipped {
  @include text-truncate;
}

// line-clamp
@for $i from 0 through 5 {
  .as-a-text--line-clamp-#{$i} {
    @include line-clamp($i);
  }
}

// html reset elements
.as-a-text--html p {
  margin: 0;
}


// --product-technical-features
.as-a-text--product-technical-features-odd,
.as-a-text--product-technical-features-even {
  width: 100%;

  ul {
    margin: 0;
    padding: 0;
    list-style: none;
  }

  li {
    padding: $a-text--product-technical-features--mobile--padding;

    @include mq($from: tablet-small) {
      padding: $a-text--product-technical-features--padding;
    }
  }
}

.as-a-text--product-technical-features-odd {
  li:nth-child(odd) {
    background: $a-text--product-technical-features__li--nth-child--background;
  }
}

.as-a-text--product-technical-features-even {
  li:nth-child(even) {
    background: $a-text--product-technical-features__li--nth-child--background;
  }
}

// --product-details
.as-a-text--product-details {
  li:not(:last-child) {
    margin-bottom: $a-text--product-details__li--margin-bottom;
  }
}

// --product-technical-description
.as-a-text--product-technical-description {
  ul {
    margin: 0;
    padding-left: $a-text--product-technical-description__ul--padding-left;
  }

  li:not(:last-child) {
    margin-bottom: $a-text--product-technical-description__li--margin-bottom;
  }
}
