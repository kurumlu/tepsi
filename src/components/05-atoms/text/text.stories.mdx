import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { mockText, mockHtml } from './text.mock';
import { Text, TextSize, TextType, TextColor, TextAlign, TextFeature } from './Text';

<Meta
  title="Design system/Patterns/Atoms/Text"
  component={Text}
/>

# Text

<Canvas>
  <Story name="Default">
    <Text>{mockText}</Text>
  </Story>
</Canvas>

---

## Available props

<ArgsTable for={Text} />

---

### Text

Pass Children to set the text that gets displayed.

```javascript
<Text>...</Text>
```

<Canvas>
  <Story name="Text">
    <Text>{mockText}</Text>
  </Story>
</Canvas>

---

### Size

Use the `size` prop to set a size. Uses the `TextSize` enum.

| TextSize  |
|-----------|
| `XLARGE`  |
| `LARGE`   |
| `MEDIUM`  |
| `SMALL`   |
| `XSMALL`  |
| `INHERIT` |

```javascript
<Text size={...}>...</Text>
```

#### XLarge

```javascript
<Text size={TextSize.XLARGE}>...</Text>
```

<Canvas>
  <Story name="Size - xlarge">
    <Text size={TextSize.XLARGE}>{mockText}</Text>
  </Story>
</Canvas>

#### Large

```javascript
<Text size={TextSize.LARGE}>...</Text>
```

<Canvas>
  <Story name="Size - large">
    <Text size={TextSize.LARGE}>{mockText}</Text>
  </Story>
</Canvas>

#### Medium

```javascript
<Text size={TextSize.MEDIUM}>...</Text>
```

<Canvas>
  <Story name="Size - medium">
    <Text size={TextSize.MEDIUM}>{mockText}</Text>
  </Story>
</Canvas>

#### Small

```javascript
<Text size={TextSize.SMALL}>...</Text>
```

<Canvas>
  <Story name="Size - small">
    <Text size={TextSize.SMALL}>{mockText}</Text>
  </Story>
</Canvas>

#### XSmall

```javascript
<Text size={TextSize.XSMALL}>...</Text>
```

<Canvas>
  <Story name="Size - xsmall">
    <Text size={TextSize.XSMALL}>{mockText}</Text>
  </Story>
</Canvas>

#### Inherit

```javascript
<Text size={TextSize.INHERIT}>...</Text>
```

<Canvas>
  <Story name="Size - inherit">
    <Text size={TextSize.INHERIT}>{mockText}</Text>
  </Story>
</Canvas>

---

### Type

Use the `type` prop to set a type. Uses the `TextType` enum.

| TextType              |
|-----------------------|
| `EXTRA_LARGE_CONTENT` |
| `EXTRA_CONTENT`       |
| `CONTENT`             |
| `SUB_CONTENT`         |

```javascript
<Text type={...}>...</Text>
```
#### Extra Large Content

```javascript
<Text size={TextType.EXTRA_LARGE_CONTENT}>...</Text>
```

<Canvas>
  <Story name="Type - extra large content">
    <Text size={TextType.EXTRA_LARGE_CONTENT}>{mockText}</Text>
  </Story>
</Canvas>

#### Extra Content

```javascript
<Text size={TextType.EXTRA_CONTENT}>...</Text>
```

<Canvas>
  <Story name="Type - extra content">
    <Text size={TextType.EXTRA_CONTENT}>{mockText}</Text>
  </Story>
</Canvas>

#### Content

```javascript
<Text size={TextType.CONTENT}>...</Text>
```

<Canvas>
  <Story name="Type - content">
    <Text size={TextType.CONTENT}>{mockText}</Text>
  </Story>
</Canvas>

#### Sub Content

```javascript
<Text size={TextType.SUB_CONTENT}>...</Text>
```

<Canvas>
  <Story name="Type - sub content">
    <Text size={TextType.SUB_CONTENT}>{mockText}</Text>
  </Story>
</Canvas>

#### Title

```javascript
<Text size={TextType.TITLE}>...</Text>
```

<Canvas>
  <Story name="Type - title">
    <Text size={TextType.TITLE}>{mockText}</Text>
  </Story>
</Canvas>

---

### Color

Use the `color` prop to set a color. Uses the `TextColor` enum.

| TextColor   |
|-------------|
| `NEGATIVE`  |
| `POSITIVE`  |
| `SUBTLE`    |
| `DISCOUNT`  |
| `INHERIT`   |
| `INFO`      |
| `NOTICE`    |

```javascript
<Text color={...}>...</Text>
```

#### Negative

```javascript
<Text color={TextColor.NEGATIVE}>...</Text>
```

<Canvas>
  <Story name="Color - negative">
    <Text color={TextColor.NEGATIVE}>{mockText}</Text>
  </Story>
</Canvas>

#### Positive

```javascript
<Text color={TextColor.POSITIVE}>...</Text>
```

<Canvas>
  <Story name="Color - positive">
    <Text color={TextColor.POSITIVE}>{mockText}</Text>
  </Story>
</Canvas>

#### Subtle

```javascript
<Text color={TextColor.SUBTLE}>...</Text>
```

<Canvas>
  <Story name="Color - subtle">
    <Text color={TextColor.SUBTLE}>{mockText}</Text>
  </Story>
</Canvas>

#### Discount

```javascript
<Text color={TextColor.DISCOUNT}>...</Text>
```

<Canvas>
  <Story name="Color - discount">
    <Text color={TextColor.DISCOUNT}>{mockText}</Text>
  </Story>
</Canvas>

#### Inherit

```javascript
<Text color={TextColor.INHERIT}>...</Text>
```

<Canvas>
  <Story name="Color - inherit">
    <span style={{color: 'purple'}}>
      <Text color={TextColor.INHERIT}>{mockText}</Text>
    </span>
  </Story>
</Canvas>

#### Info

```javascript
<Text color={TextColor.INFO}>...</Text>
```

<Canvas>
  <Story name="Color - info">
    <Text color={TextColor.INFO}>{mockText}</Text>
  </Story>
</Canvas>

#### Notice

```javascript
<Text color={TextColor.NOTICE}>...</Text>
```

<Canvas>
  <Story name="Color - notice">
    <Text color={TextColor.NOTICE}>{mockText}</Text>
  </Story>
</Canvas>

---

### Strike

Use the `strike` prop to add a strikethrough line.

```javascript
<Text strike>...</Text>
```

<Canvas>
  <Story name="Strike">
    <Text strike>{mockText}</Text>
  </Story>
</Canvas>

---

### Emphasized

Use the `emphasized` prop to use the emphasized font-map.


```javascript
<Text emphasized>...</Text>
```

<Canvas>
  <Story name="Emphasized">
    <Text emphasized>{mockText}</Text>
  </Story>
</Canvas>

---

### Paragraph

Use the `paragraph` prop to make it a paragraph.

```javascript
<Text paragraph>...</Text>
```

<Canvas>
  <Story name="Paragraph">
    <React.Fragment>
      <Text paragraph>{mockText}</Text>
      <Text paragraph>{mockText}</Text>
    </React.Fragment>
  </Story>
</Canvas>

---

### Inline

Use the `inline` prop to replace `div` to `span` element.

```javascript
<Text inline>...</Text>
```

<Canvas>
  <Story name="inline">
    <React.Fragment>
      <Text inline>{mockText}</Text>
      <Text inline>{mockText}</Text>
    </React.Fragment>
  </Story>
</Canvas>

---

### Alignment

Use the `align` prop to set the alignment of the text. Uses the `TextAlign` enum.

| TextAlign |
|-----------|
| `CENTER`  |
| `RIGHT`   |

```javascript
<Text align={...} paragraph>...</Text>
```

#### Left

```javascript
<Text paragraph>...</Text>
```

<Canvas>
  <Story name="Align - left">
    <Text paragraph>{mockText}</Text>
  </Story>
</Canvas>

#### Center

```javascript
<Text align={TextAlign.CENTER} paragraph>...</Text>
```

<Canvas>
  <Story name="Align - center">
    <Text align={TextAlign.CENTER} paragraph>{mockText}</Text>
  </Story>
</Canvas>

#### Right

```javascript
<Text align={TextAlign.RIGHT} paragraph>...</Text>
```

<Canvas>
  <Story name="Align - right">
    <Text align={TextAlign.RIGHT} paragraph>{mockText}</Text>
  </Story>
</Canvas>

---

### Feature

Use the `feature` prop to set a specific feature type. Uses the `TextFeature` enum.

| TextFeature                       |
|-----------------------------------|
| `PRODUCT_TECHNICAL_FEATURES_ODD`  |
| `PRODUCT_TECHNICAL_FEATURES_EVEN` |
| `PRODUCT_DETAILS`                 |
| `PRODUCT_TECHNICAL_DESCRIPTION`   |

```javascript
<Text feature={...}>...</Text>
```

#### Product technical features odd

```javascript
<Text feature={TextFeature.PRODUCT_TECHNICAL_FEATURES_ODD}>...</Text>
```

<Canvas>
  <Story name="Feature - Technical - Odd">
    <Text feature={TextFeature.PRODUCT_TECHNICAL_FEATURES_ODD}>
      <ul>
        <li>Waterdicht: Ja</li>
        <li>Model: Heren</li>
        <li>Materiaal: 100% Polyester</li>
        <li>Capuchon: Ja</li>
        <li>Pasvorm: Regular Fit</li>
        <li>Waterafstotende behandeling: Ja</li>
      </ul>
    </Text>
  </Story>
</Canvas>

#### Product technical features even

```javascript
<Text feature={TextFeature.PRODUCT_TECHNICAL_FEATURES_EVEN}>...</Text>
```

<Canvas>
  <Story name="Feature - Technical - Even">
    <Text feature={TextFeature.PRODUCT_TECHNICAL_FEATURES_EVEN}>
      <ul>
        <li>Waterdicht: Ja</li>
        <li>Model: Heren</li>
        <li>Materiaal: 100% Polyester</li>
        <li>Capuchon: Ja</li>
        <li>Pasvorm: Regular Fit</li>
        <li>Waterafstotende behandeling: Ja</li>
      </ul>
    </Text>
  </Story>
</Canvas>

#### Product details

```javascript
<Text feature={TextFeature.PRODUCT_DETAILS}>...</Text>
```

<Canvas>
  <Story name="Feature - Details">
    <Text feature={TextFeature.PRODUCT_DETAILS}>
      <ul>
        <li>Getapete naden</li>
        <li>Borstzak met waterdichte rits</li>
        <li>Uitritsbare fleece binnenjas met steekzakken</li>
      </ul>
    </Text>
  </Story>
</Canvas>

#### Product technical description

```javascript
<Text feature={TextFeature.PRODUCT_TECHNICAL_DESCRIPTION}>...</Text>
```

<Canvas>
  <Story name="Feature - Technical description">
    <Text feature={TextFeature.PRODUCT_TECHNICAL_DESCRIPTION}>
      <ul>
        <li>Getapete naden</li>
        <li>Borstzak met waterdichte rits</li>
        <li>Uitritsbare fleece binnenjas met steekzakken</li>
      </ul>
    </Text>
  </Story>
</Canvas>

---

### Clipped text

Use the `isClipped` prop to make the text clip and show an ellipsis when it overflows it's parent.

```javascript
<Text isClipped>...</Text>
```

<Canvas>
  <Story name="Clipped text">
    <div style={{ width: 300 }}>
      <Text isClipped>
        {mockText}
      </Text>
    </div>
  </Story>
</Canvas>

---

### Line Clamp text

Use the `lineClamp` prop to truncate text at a specific number of lines.

```javascript
<Text lineClamp={2}>...</Text>
```

<Canvas>
  <Story name="Line Clamp text">
    <Text lineClamp={2}>{mockText.repeat(2)}</Text>
  </Story>
</Canvas>

---

### Uppercase

Use the `isUppercase` prop to make the text uppercase.

```javascript
<Text isUppercase>...</Text>
```

<Canvas>
  <Story name="Uppercase">
    <Text isUppercase>
      {mockText}
    </Text>
  </Story>
</Canvas>

---

### HTML

Use the `html` prop to set inner html.

```javascript
<Text html="<div/>">...</Text>
```

<Canvas>
  <Story name="HTML">
    <Text html={mockHtml} />
  </Story>
</Canvas>

---

### Other

Other props that can be added:
- `className`
- `dataQA`
- `withGlossary` - adds `GLOSSARY_HIGHLIGHT_CLASS` className. Needs for tooltip functionality
- `itemProp`
- `itemContent`

---

## Examples

Some examples of texts.

---

### Links

An anchor inside of a text atom (without tepsi classes: `sd-...`) will be styled to look like a default link.

<Canvas>
  <Story name="Contains link">
    <Text size={TextSize.XLARGE} color={TextColor.POSITIVE}>
      We believe there’s no substitute for <a href="#">experience</a>.
    </Text>
  </Story>
</Canvas>
