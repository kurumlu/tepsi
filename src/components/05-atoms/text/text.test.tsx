import React from 'react';
import { shallow, ShallowWrapper, mount } from 'enzyme';

import { Text, TextProps, TextColor, TextSemanticTag } from './Text';
import { mockText, MockChildren } from './text.mock';

describe('<Text />', () => {
  const component: ShallowWrapper<TextProps> = shallow(<Text inline></Text>);

  it('renders <Text /> component', () => {
    expect(component).toMatchSnapshot('default');
  });

  it('renders <Text /> with HTML if "html" prop passed component', () => {
    const componentWithHTML: ShallowWrapper<TextProps> = shallow(
      <Text html="<b>bold</b>" />
    );
    expect(componentWithHTML).toMatchSnapshot('text with html');
  });

  it(`adds classname 'sd-a-text--${TextColor.NEGATIVE}' if "color" props passed`, () => {
    const componentWithColor: ShallowWrapper<TextProps> = shallow(
      <Text inline color={TextColor.NEGATIVE}>
        {mockText}
      </Text>
    );

    expect(componentWithColor).toMatchSnapshot('color');
    expect(
      componentWithColor.hasClass(`sd-a-text--${TextColor.NEGATIVE}`)
    ).toBeTruthy();
  });

  it(`adds ${TextSemanticTag.DEL} tag if "strike" prop was passed`, () => {
    const componentWithStrike: ShallowWrapper<TextProps> = shallow(
      <Text inline strike>
        {mockText}
      </Text>
    );

    expect(componentWithStrike).toMatchSnapshot('strike');
    expect(componentWithStrike.find(TextSemanticTag.DEL)).toBeTruthy();
  });

  it(`adds ${TextSemanticTag.STRONG} tag if "emphasized" prop was passed`, () => {
    const componentWithEmphasized: ShallowWrapper<TextProps> = shallow(
      <Text inline emphasized>
        {mockText}
      </Text>
    );

    expect(componentWithEmphasized).toMatchSnapshot('emphasized');
    expect(componentWithEmphasized.find(TextSemanticTag.STRONG)).toBeTruthy();
  });

  it(`replace span tag to ${TextSemanticTag.PARAGRAPH} if "paragraph" prop was passed`, () => {
    const componentWithParagraph: ShallowWrapper<TextProps> = shallow(
      <Text paragraph>{mockText}</Text>
    );

    expect(componentWithParagraph).toMatchSnapshot('paragraph');
    expect(
      componentWithParagraph.exists(TextSemanticTag.PARAGRAPH)
    ).toBeTruthy();
    expect(
      componentWithParagraph
        .find(TextSemanticTag.PARAGRAPH)
        .hasClass('sd-a-text')
    ).toBeTruthy();
  });

  it(`renders <Text /> component with children`, () => {
    const componentWithChildren = mount(
      <Text>
        <MockChildren />
      </Text>
    );

    expect(componentWithChildren).toMatchSnapshot('children');
    expect(componentWithChildren.find('ul')).toHaveLength(1);
  });

  it('should have "sd-a-text--clipped" class if isClipped prop was passed', () => {
    const componentIsClipped: ShallowWrapper<TextProps> = shallow(
      <Text isClipped inline>
        {mockText}
      </Text>
    );

    expect(componentIsClipped.hasClass('sd-a-text--clipped')).toBeTruthy();
  });

  it('should have "sd-a-text--line-clamp-X" class if lineClamp prop was passed', () => {
    const lineClamp = 2;
    const componentLineClamp: ShallowWrapper<TextProps> = shallow(
      <Text lineClamp={lineClamp}>{mockText}</Text>
    );

    expect(
      componentLineClamp.hasClass(`sd-a-text--line-clamp-${lineClamp}`)
    ).toBeTruthy();
  });
});
