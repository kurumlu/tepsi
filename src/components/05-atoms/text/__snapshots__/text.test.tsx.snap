// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> adds classname 'as-a-text--negative' if "color" props passed: color 1`] = `
<span
  className="as-a-text as-a-text--negative"
>
  We believe there’s no substitute for experience. That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</span>
`;

exports[`<Text /> adds del tag if "strike" prop was passed: strike 1`] = `
<span
  className="as-a-text"
>
  <del>
    We believe there’s no substitute for experience. That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
  </del>
</span>
`;

exports[`<Text /> adds strong tag if "emphasized" prop was passed: emphasized 1`] = `
<span
  className="as-a-text"
>
  <strong>
    We believe there’s no substitute for experience. That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
  </strong>
</span>
`;

exports[`<Text /> renders <Text /> component with children: children 1`] = `
<Text>
  <div
    className="as-a-text"
  >
    <MockChildren>
      <ul>
        <li>
          Test text 1
        </li>
        <li>
          Test text 2
        </li>
        <li>
          Test text 3
        </li>
      </ul>
    </MockChildren>
  </div>
</Text>
`;

exports[`<Text /> renders <Text /> component: default 1`] = `
<span
  className="as-a-text"
/>
`;

exports[`<Text /> renders <Text /> with HTML if "html" prop passed component: text with html 1`] = `
<div
  className="as-a-text as-a-text--html"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>bold</b>",
    }
  }
/>
`;

exports[`<Text /> replace span tag to p if "paragraph" prop was passed: paragraph 1`] = `
<p
  className="as-a-text"
>
  We believe there’s no substitute for experience. That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</p>
`;
